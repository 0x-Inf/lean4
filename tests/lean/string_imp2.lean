def f (s : string) : string :=
s ++ " " ++ s

#eval "hello" ++ "hello"
#eval f "hello"
#eval (f "αβ").length
#eval "hello".to_list
#eval "αβ".to_list
#eval "".to_list
#eval "αβγ".to_list
#eval "αβγ".fold [] (λ r c, c::r)
#eval "".fold 0 (λ r c, r+1)
#eval "αβγ".fold 0 (λ r c, r+1)
#eval "αβγ".mk_iterator.1
#eval "αβγ".mk_iterator.next.1
#eval "αβγ".mk_iterator.next.next.1
#eval "αβγ".mk_iterator.next.2
#eval "αβ".1
