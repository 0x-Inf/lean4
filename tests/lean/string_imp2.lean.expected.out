"hellohello"
"hello hello"
5
['h', 'e', 'l', 'l', 'o']
['α', 'β']
[]
['α', 'β', 'γ']
['γ', 'β', 'α']
0
3
[]
['α']
['β', 'α']
['β', 'γ']
